src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1559,8 +1559,13 @@ private static ParameterSymbol GetParameter(int argIndex, MemberAnalysisResult r
                                                    ref useSiteDiagnostics,
                                                    out okToDowngradeToNeither);
 
-                var type1Normalized = type1.NormalizeTaskTypes(Compilation);
-                var type2Normalized = type2.NormalizeTaskTypes(Compilation);
+                var type1Normalized = type1;
+                var type2Normalized = type2;
+                if (!_binder.InAttributeArgument)
```

Comment:
We should include a comment on why this check is needed and valid. Perhaps something like:
Avoid normalizing task types in attribute arguments to avoid cycles binding attributes on task-like types. Normalizing is only necessary to differentiate async lambdas which are not supported in attribute arguments.